eval/corpus/cpp/ble_collection_mutex.cpp

```
@@ -0,0 +1,710 @@
+
+/**
+ * @brief Add a device configuration or replace an existing one with the same id.
+ *
+ * If the provided config has a non-empty alias, any existing configs with the same alias but a different id
+ * are removed and their ids are scheduled for deletion (deletion occurs after the function returns).
+ *
+ * @param config DeviceConfig to add or use to replace an existing entry with the same `id`.
+ * @return true if a new configuration was added, false if an existing configuration was replaced.
+ */
+bool addOrReplace(DeviceConfig config) {
+    if (xSemaphoreTake(deviceConfigMutex, MAX_WAIT) != pdTRUE)
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)